garrysmod/lua/includes/extensions/table.lua

```diff
@@ -698,3 +698,22 @@ function table.GetKeys( tab )
 	return keys
 
 end
+
+--[[---------------------------------------------------------
+	Name: table.Where( table a, boolean condition, *optional ipairs/pairs )
+	Desc: A simple Where method like c#
+-----------------------------------------------------------]]
+
+function table.Where(a, b, iterator)
+    local c = {}
+
+    iterator = iterator or pairs
+    for i, v in iterator(a or {}) do
+        local res = b(i, v)
+        if (res) then
+            c[i] = v

```

Comment:
If you use this in your own code, make sure you correct this to `c[#c+1]` otherwise, you'll end up with nil holes in your array.

Reply:
> If you use this in your own code, make sure you correct this to `c[#c+1]` , otherwise you'll end up with nil holes in your array.

It basically just filters the table, it's not supposed to return a sequential array. Maybe `table.Filter` will be a better name for it.

Reply:
It is supposed to return a sequential array

Reply:
I took a glance at the C# method. For a dictionary, the closest type to a Lua table, `Where` returns an iterator over a collection of key-value pairs. https://rextester.com/QDJN89112